pkg/controller/endpoint/endpoints_controller.go

```diff
@@ -393,17 +393,18 @@ func (e *EndpointController) syncService(key string) error {
 		return err
 	}
 	service, err := e.serviceLister.Services(namespace).Get(name)
-	if err != nil {
+	if err != nil && errors.IsNotFound(err) {
```

Comment:
@deads2k i'm actually not sure if a Lister can ever return something else than 404 as error...

Reply:
Should we check that we actually own the endpoint before deleting? If it is an external service, we don't.

Reply:
@sttts if endpoints have ownerRefs we don't have to do this at all I guess (GC should remove endpoints).

Reply:
right, even better.